packages/turf-clusters-kmeans/index.js

```diff
@@ -27,37 +23,36 @@ var featureCollection = helpers.featureCollection;
  * var clustered = turf.clustersKmeans(points, numberOfClusters);
  *
  * //addToMap
- * var addToMap = featureCollection(clustered.points);
+ * var addToMap = clustered;
  */
 module.exports = function (points, numberOfClusters) {
     // Input validation
     collectionOf(points, 'Point', 'Input must contain Points');
+
     // Default Params
     var count = points.features.length;
-    if (numberOfClusters > count) throw new Error('numberOfClusters can\'t be grated than the number of points');
     numberOfClusters = numberOfClusters || Math.round(Math.sqrt(count / 2));
 
+    // numberOfClusters can't be greater than the number of points
+    // fallbacks to count
+    if (numberOfClusters > count) numberOfClusters = count;
+
+    // Clone points to prevent any mutations
+    points = clone(points, true);
```

Comment:
We could always add a `mutate` parameter (to allow input mutations).

It will significantly increase the performance if the user doesn't mind having the input be added.

We might be able to do handle this without the use of clone, but as a safe measure it was added.

## Benchmark results

**without `clone`**

```
fiji x 260,987 ops/sec ±1.03% (87 runs sampled)
many-points x 173 ops/sec ±7.86% (77 runs sampled)
points-with-properties x 245,562 ops/sec ±1.54% (86 runs sampled)
points1 x 71,572 ops/sec ±0.86% (88 runs sampled)
points2 x 40,784 ops/sec ±1.06% (86 runs sampled)
```
**with `clone` (~6 times slower)**
```
fiji x 41,566 ops/sec ±1.18% (89 runs sampled)
many-points x 145 ops/sec ±2.42% (79 runs sampled)
points-with-properties x 38,453 ops/sec ±1.22% (83 runs sampled)
points1 x 12,598 ops/sec ±1.09% (91 runs sampled)
points2 x 8,124 ops/sec ±0.97% (89 runs sampled)
```